caput/mpiarray.py

```diff
@@ -245,6 +236,71 @@ class MPIArray(np.ndarray):
         The dimension to distribute the array across.
     """
 
+    def __getitem__(self, v):
+        # Return an MPIArray view
+
+        if not isinstance(v, tuple):
+            return super().__getitem__(v)
+
+        try:
+            dist_axis_index = v[self.axis]
```

Comment:
I don't think this works. What if I run this through it?
```
dist_arr = mpiarray.MPIArray((4,4), axis=1, comm=comm)
a = dist_arr[0]
```
I think it'll fail with an IndexError.